aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -1061,18 +1103,30 @@ public Map<String, String> httpClientProperties() {
     return httpClientProperties;
   }
 
+  public String clientRegion() {
+    return clientRegion;
+  }
+
+  public void setClientRegion(String clientRegion) {
+    this.clientRegion = clientRegion;
+  }
+
   /**
    * Configure the credentials for an S3 client.
    *
    * <p>Sample usage:
    *
    * <pre>
-   *     S3Client.builder().applyMutation(awsProperties::applyS3CredentialConfigurations)
+   *     S3Client.builder().applyMutation(awsProperties::applyCredentialConfigurations)
    * </pre>
    */
-  public <T extends S3ClientBuilder> void applyS3CredentialConfigurations(T builder) {
-    builder.credentialsProvider(
-        credentialsProvider(s3AccessKeyId, s3SecretAccessKey, s3SessionToken));
+  public <T extends AwsClientBuilder> void applyCredentialConfigurations(T builder) {
```

Comment:
I think we need 2 methods here, 1 for `applyClientCredentialConfigurations`, 1 for `applyS3CredentialConfigurations`, because `s3AccessKeyId, s3SecretAccessKey, s3SessionToken` are S3 only credential settings that are not applicable to other clients.